### PR TITLE
[code-completion] Handle nested keypath dynamic lookup correctly

### PR DESCRIPTION
**Pull request by benlangmuir**
If the root type of the KeyPath also has dynamic member lookup, we need
to look through it.

rdar://problem/50450037

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

File: lib/Sema/LookupVisibleDecls.cpp [hunk 1]
```diff
@@ -913,7 +921,28 @@ static void lookupVisibleDynamicMemberLookupDecls(
     Type baseType, KeyPathDynamicMemberConsumer &consumer,
     const DeclContext *dc, LookupState LS, DeclVisibilityKind reason,
     LazyResolver *typeResolver, GenericSignatureBuilder *GSB,
-    VisitedSet &visited) {
+    VisitedSet &visited, llvm::DenseSet<TypeBase *> &seenDynamicLookup);
+
+static void lookupVisibleMemberAndDynamicMemberDecls(
```
Comment (rintaro) [inline lib/Sema/LookupVisibleDecls.cpp]: Function names are getting confusing. Could you add doc-comments to added functions in this PR and #24520 ?

Comment (benlangmuir) [reply lib/Sema/LookupVisibleDecls.cpp]: This function is so we have a place to introduce recursion for dynamic lookup.  I wanted to put all the calls in `lookupVisibleMemberDeclsImpl`, but that is already recursive in the wrong way for doing dynamic member lookup :-(  I'll think about this some more and see if I can come up with better names.  Will add doc comments as requested too.

**Comment by rintaro** [comment]
Implementation itself looks good to me.

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

**Pull request merged**